### PR TITLE
Cnwankwo/together_mode_api

### PR DESCRIPTION
**Pull request by cn0151**
# What
The changes here contains
- APIs needed for Together Mode feature
 - - onStartTogetherMode - Only Teams User should be able to start TogetherMode
 - - onCreateTogetherModeStreamView - Method needed to create the togetherMode stream
 - - onSetTogetherModeSceneSize - Method for recalculating the seating positions when a screen size changes
 - - onDisposeTogetherModeStreamViews - Method for disposing the together mode stream

-    Props needed for togetherMode component

# Why
This is part of integrating togetherMode as part of web ui

# How Tested
Tested locally

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-components/src/types/TogetherModeTypes.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+export interface TogetherModeStreams {
+  mainVideoStream?: TogetherModeVideoStream;
+}
+
+/* @conditional-compile-remove(together-mode) */
+/**
+ * Interface representing the position of a participant in Together Mode.
+ * @beta
+ */
+export interface TogetherModeParticipantPosition {
```
Comment (cn0151) [inline packages/react-components/src/types/TogetherModeTypes.ts:50]: Update to Record<ParticipantId, seatingPosition>

File: packages/calling-component-bindings/src/handlers/createTeamsCallHandlers.ts [hunk 2]
```diff
@@ -125,6 +127,20 @@ export const createDefaultTeamsCallingHandlers = memoizeOne(
...
+      /* @conditional-compile-remove(together-mode) */
+      onStartTogetherMode: async (): Promise<void> => {
+        if (!call) {
+          return;
+        }
+        const callState = callClient.getState().calls[call.id];
+        if (!callState) {
+          return;
+        }
+        if (!callState.togetherMode.isActive) {
```
Comment (JamesBurnside) [inline packages/calling-component-bindings/src/handlers/createTeamsCallHandlers.ts:140]: Is this check necessary? We should make `togetherModeFeature.start()` to be idempotent meaning if it's already started then `.start()` will noop.

That would make this function reduce down to:
```ts
onStartTogetherMode: async (): Promise<void> => {
  return call?.feature(Features.TogetherMode)?.start();
}
```

File: packages/react-components/src/types/TogetherModeTypes.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+export interface TogetherModeStreamViewResult {
+  mainVideoView?: CreateVideoStreamViewResult;
+}
+
+/* @conditional-compile-remove(together-mode) */
+/**
+ * Represents a video stream in Together Mode.
+ * @beta
+ */
+export interface TogetherModeVideoStream {
```
Comment (JamesBurnside) [inline packages/react-components/src/types/TogetherModeTypes.ts:21]: Does/should together mode have an `isReceiving` or `scalingMode` property?

**Review by JamesBurnside — COMMENTED** [review]
Add details to PR title and PR description

File: packages/calling-component-bindings/src/handlers/createTeamsCallHandlers.ts (hunk 2 above)
Comment (cn0151) [reply packages/calling-component-bindings/src/handlers/createTeamsCallHandlers.ts:140]: I created a user story to look into this on the SDK side

File: packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts [hunk 4]
```diff
@@ -609,6 +610,54 @@ export interface CallAdapterCallOperations {
...
+   *
+   * @remarks
+   * This method is implemented for composite
+   *
+   * @param width - Width of the container
+   * @param height - Height of the container
+   *
+   * @beta
+   */
+  setTogetherModeSceneSize(width: number, height: number): void;
```
Comment (dmceachernmsft) [inline packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts:645]: Is this something that we would want to give contoso access to? I think that we could have this set internally so that our resize observers are setting this.

File: samples/CallWithChat/public/index.html [hunk 5]
```diff
@@ -22,8 +22,14 @@
       document.head.appendChild(script);
     </script>
     <style>
-      body { display: flex; flex-direction: column; height: auto; }
-      #Root { flex: 1; }
+      body {
```
Comment (dmceachernmsft) [inline samples/CallWithChat/public/index.html:25]: Are these changes needed?

Comment (cn0151) [reply samples/CallWithChat/public/index.html:25]: I did not make this change. I believe this happened because rush prettier runs every time I commit

File: packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts (hunk 4 above)
Comment (cn0151) [reply packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts]: For now I removed it and would look into it in my next PR

File: packages/calling-component-bindings/src/handlers/createCommonHandlers.ts [hunk 6]
```diff
@@ -817,6 +897,14 @@ export const createDefaultCommonCallingHandlers = memoizeOne(
...
       onAcceptCall: notImplemented,
       onRejectCall: notImplemented,
+      /* @conditional-compile-remove(together-mode) */
+      onCreateTogetherModeStreamView,
+      /* @conditional-compile-remove(together-mode) */
+      onStartTogetherMode: notImplemented,
+      /* @conditional-compile-remove(together-mode) */
+      onDisposeTogetherModeStreamViews,
+      /* @conditional-compile-remove(together-mode) */
+      onSetTogetherModeSceneSize,
```
Comment (dmceachernmsft) [inline packages/calling-component-bindings/src/handlers/createCommonHandlers.ts:907]: should this also be included in the type above?

**Review by dmceachernmsft — APPROVED** [review]
Going to approve to unblock but I think the seating mode API is missing still please add back before completing the pullrequest

File: packages/calling-component-bindings/src/handlers/createCommonHandlers.ts (hunk 6 above)
Comment (cn0151) [reply packages/calling-component-bindings/src/handlers/createCommonHandlers.ts:907]: onSetTogetherModeSceneSize is a callback just like onStartTogetherMode. So far, we don't need it as a type

**Review by JamesBurnside — APPROVED** [review]
